tests/unit/zhmcclient/test_resource.py

```diff
@@ -642,3 +643,778 @@ def test_get_properties_local(
+            'allowed-capacity': 32,
+            'maximum-total-capacity': 32,
+            'physical-channel-status': 'operating',
+            'maximum-transmission-unit-size': 56,
+        })
+
+        # Port properties that will be auto-set:
+        # - element-uri
+        # Properties in parent adapter that will be auto-set:
+        # - network-port-uris
+        faked_hs.ports.add({
+            'element-id': 'fake-port1-oid',
+            'parent': faked_hs.uri,
+            'class': 'network-port',
+            'index': 0,
+            'name': 'fake-port1-name',
+            'description': 'Hipersocket #1 Port #1',
+        })
+        return faked_hs
+
+    def add_standard_partition(self):
+        """
+        Add a standard partition.
+
+        Partition resources support the 'properties' query parameter.
+        """
+
+        # Partition properties that will be auto-set:
+        # - object-uri
+        faked_part = self.faked_cpc.partitions.add({
+            'object-id': self.RESOURCE_OID,
+            'parent': self.faked_cpc.uri,
+            'class': 'partition',
+            'name': self.RESOURCE_NAME,
+            'description': self.RESOURCE_DESC,
+            'status': 'stopped',
+        })
+        return faked_part
+
+    @pytest.mark.parametrize(
+        # Indicates whether to delete the resource before test
+        "delete", [False, True])
+    def test_pull_full_properties(self, delete):
+        """
+        Test BaseResource.pull_full_properties().
+        """
+        faked_res = self.add_standard_hipersocket()
+        res_mgr = self.cpc.adapters
+        resource = res_mgr.find(name=faked_res.name)
+        propnames_find = set(resource.properties.keys())
+        assert resource.full_properties is False
+
+        if delete:
+            resource.delete()
+
+            with pytest.raises(HTTPError):
+
+                # The code to be tested
+                resource.pull_full_properties()
+
+        else:
+
+            # The code to be tested
+            resource.pull_full_properties()
+
+            assert resource.full_properties is True
+            propnames_pull = set(resource.properties.keys())
+            assert propnames_pull.issuperset(propnames_find)
+            assert dict(resource.properties) == faked_res.properties
+
+    TESTCASES_PULL_PROPERTIES = [
+        # Testcases for test_pull_properties().
+        # Each list item is a tuple defining a testcase in the following format:
+        # - input_kwargs: keyword input parameters for the function
+        # - delete: Indicates whether to delete the resource before test
+        # - supports_properties: Resource supports 'properties' query parm
+        # - exp_propnames: Expected property names in resource after test
+        # - exp_full_props: Expected value for full_properties attribute
+        # - exp_exc_type: Expected type of exception, or None for success
+
+        (
+            dict(properties=None),
```

Comment:
would it make sense to add a comment to each testcase to explain what it is testing?

Reply:
We often have a "description" item in such testcases, but this time I skipped that.
You are right in that we should have it. I'll add it.

Reply:
Since PR #1292 already merged the content of this PR, the change needs to be done in a future PR. I have created issue TBD for that.